### PR TITLE
sql-statements: add doc of lock tables and unlock tables

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Close https://github.com/pingcap/docs/issues/8632

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/13301
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by crazycs520 — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+# LOCK TABLES and UNLOCK TABLES
+
+> **Warning:**
+>
+> `LOCK TABLES` and `UNLOCK TABLES` are experimental features for the current version. It is not recommended to use it in the production environment.
+
+TiDB enables client sessions to acquire table locks for the purpose of cooperating with other sessions for access to tables, or to prevent other sessions from modifying tables. A session can acquire or release locks only for itself. One session cannot acquire locks for another session or release locks held by another session.
+
+`LOCK TABLES` acquires table locks for the current client session. Table locks can be acquired for common tables on the condition that you have the `LOCK TABLES` and `SELECT` privilege for each object to be locked.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:14]:
```suggestion
`LOCK TABLES` acquires table locks for the current client session. If you have the `LOCK TABLES` and `SELECT` privilege for each object to be locked, you can acquire table locks for common tables .
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+`UNLOCK TABLES` explicitly releases any table locks held by the current session. `LOCK TABLES` implicitly releases any table locks held by the current session before acquiring new locks.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:16]:
```suggestion
`UNLOCK TABLES` explicitly releases any table locks held by the current session. `LOCK TABLES` implicitly releases all table locks held by the current session before acquiring new locks.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+A table lock protects only against inappropriate reads or writes by other sessions. A session holding a `WRITE` lock can perform table-level operations such as `DROP TABLE` or `TRUNCATE TABLE`.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:18]:
```suggestion
A table lock protects against reads or writes by other sessions. A session that holds a `WRITE` lock can perform table-level operations such as `DROP TABLE` or `TRUNCATE TABLE`.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+- The session that holds this lock can read and write the table.
+- Only the session that holds this lock can access the table. No other session can access it until the lock is released.
+
+`WRITE LOCAL` lock:
+
+- The session that holds this lock can read and write the table.
+- Only the session that holds this lock can access the table. Other sessions can read the table, but not write it.
+
+If the `LOCK TABLES` statement must wait due to locks held by other sessions on any of the tables, an error is returned upon the execution of this statement, for example:
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:61]:
```suggestion
If the lock that the `LOCK TABLES` statement needs is held by another session, the `LOCK TABLES` statement must wait, and an error is returned upon the execution of this statement, for example:
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+You cannot acquire the same table lock multiple times in a single `LOCK TABLES` statement.
+
+```sql
+> LOCK TABLE t WRITE, t READ;
+ERROR 1066 (42000): Not unique table/alias: 't'
+```
+
+## Table lock release
+
+When the table locks held by a session are released, they are all released at the same time. A session can release its locks explicitly, or locks may be released implicitly under certain conditions.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:79]:
```suggestion
When the table locks held by a session are released, they are all released at the same time. A session can release its locks explicitly or implicitly.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+`WRITE LOCAL` lock:
+
+- The session that holds this lock can read and write the table.
+- Only the session that holds this lock can access the table. Other sessions can read the table, but not write it.
+
+If the `LOCK TABLES` statement must wait due to locks held by other sessions on any of the tables, an error is returned upon the execution of this statement, for example:
+
+```sql
+> lock table t1 read;
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:64]:
```suggestion
> LOCK TABLE t1 READ;
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+> LOCK TABLE t WRITE, t READ;
+ERROR 1066 (42000): Not unique table/alias: 't'
+```
+
+## Table lock release
+
+When the table locks held by a session are released, they are all released at the same time. A session can release its locks explicitly, or locks may be released implicitly under certain conditions.
+
+- A session can release its locks explicitly with `UNLOCK TABLES`.
+- If a session issues a `LOCK TABLES` statement to acquire a lock while already holding locks, its existing locks are released implicitly before the new locks are granted.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:82]:
```suggestion
- If a session issues a `LOCK TABLES` statement to acquire a lock while already holding locks, its existing locks are released implicitly before the new locks are acquired.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+If the connection for a client session terminates, whether normally or abnormally, TiDB implicitly releases all table locks held by the session. If the client reconnects, the locks are no longer in effect. For this reason, it is not recommended to enable auto-reconnect on the client. With auto-reconnect enabled, the client is not notified if reconnect occurs but any table locks or current transaction are lost. With auto-reconnect disabled, if the connection drops, an error occurs for the next statement issued. The client can detect the error and take appropriate action such as reacquiring the locks or redoing the transaction.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:84]:
```suggestion
If the connection for a client session terminates, whether normally or abnormally, TiDB implicitly releases all table locks held by the session. If the client reconnects, the locks are no longer in effect. For this reason, it is not recommended to enable auto-reconnection on the client. If you enable auto-reconnection, the client is not notified when reconnection occurs, and all table locks or current transactions are lost. By contrast, with auto-reconnection disabled, if the connection drops, an error occurs when the next statement is issued. The client can detect the error and take appropriate action such as reacquiring the locks or redoing the transaction.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+## Table-locking restrictions and conditions
+
+You can safely use `KILL` to terminate a session that is waiting for a table lock.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:88]:
```suggestion
You can safely use `KILL` to terminate a session that holds a table lock.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+You cannot acquire table locks on tables in the `INFORMATION_SCHEMA`, `PERFORMANCE_SCHEMA`, `METRICS_SCHEMA`, and `mysql` databases.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:90]:
```suggestion
You cannot acquire table locks on tables in the following databases:

- `INFORMATION_SCHEMA`
- `PERFORMANCE_SCHEMA`
- `METRICS_SCHEMA` 
- `mysql`
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+## MySQL compatibility
+
+### Table lock acquisition
+
+- In TiDB, if session A has already held a table lock, an error is returned if session B attempts to write to the table. In MySQL, the write request of session B is blocked until session A releases the table lock. Table locking requests from other sessions are blocked until the current session releases the `WRITE` lock.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:96]:
```suggestion
- In TiDB, if session A has already held a table lock, an error is returned if session B attempts to write to the table. In MySQL, the write request of session B is blocked until session A releases the table lock. Requests for locking the table from other sessions are blocked until the current session releases the `WRITE` lock.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+> lock table t1 read;
+ERROR 8020 (HY000): Table 't1' was locked in WRITE by server: f4799bcb-cad7-4285-8a6d-23d3555173f1_session: 2199023255959
+```
+
+The preceding error message indicates that the session with ID `2199023255959` in TiDB `f4799bcb-cad7-4285-8a6d-23d3555173f1` already holds a `WRITE` lock on table `t1`. Therefore, the current session cannot acquire a `READ` lock on table `t1`.
+
+You cannot acquire the same table lock multiple times in a single `LOCK TABLES` statement.
+
+```sql
+> LOCK TABLE t WRITE, t READ;
```
Comment (shichun-0415) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:73]:
```suggestion
> LOCK TABLES t WRITE, t READ;
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+
+> **Warning：**
+>
+> To enable table locks, you need to set [`enable-table-lock`](/tidb-configuration-file.md#enable-table-lock-new-in-v400) to `true` in the configuration files of all TiDB instances.
+
+## Synopsis
+
+```ebnf+diagram
+LockTablesDef
+         ::= 'LOCK' 'TABLES' TableName LockType ( ',' TableName LockType)*
```
Comment (shichun-0415) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:28]:
```suggestion
         ::= 'LOCK' ( 'TABLES' | 'TABLE' ) TableName LockType ( ',' TableName LockType)*
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+`WRITE LOCAL` lock:
+
+- The session that holds this lock can read and write the table.
+- Only the session that holds this lock can access the table. Other sessions can read the table, but not write it.
+
+If the lock that the `LOCK TABLES` statement needs is held by another session, the `LOCK TABLES` statement must wait, and an error is returned upon the execution of this statement, for example:
+
+```sql
+> LOCK TABLE t1 READ;
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:64]:
```suggestion
> LOCK TABLES t1 READ;
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+# LOCK TABLES and UNLOCK TABLES
+
+> **Warning:**
+>
+> `LOCK TABLES` and `UNLOCK TABLES` are experimental features for the current version. It is not recommended to use it in the production environment.
+
+TiDB enables client sessions to acquire table locks for the purpose of cooperating with other sessions for access to tables, or to prevent other sessions from modifying tables. A session can acquire or release locks only for itself. One session cannot acquire locks for another session or release locks held by another session.
+
+`LOCK TABLES` acquires table locks for the current client session. If you have the `LOCK TABLES` and `SELECT` privilege for each object to be locked, you can acquire table locks for common tables .
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:14]:
```suggestion
`LOCK TABLES` acquires table locks for the current client session. If you have the `LOCK TABLES` and `SELECT` privilege for each object to be locked, you can acquire table locks for common tables.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+           | 'WRITE' ('LOCAL')?
+```
+
+## Acquire table locks
+
+You can acquire table locks within the current session by using the `LOCK TABLES` statement. The following lock types are available:
+
+`READ` lock:
+
+- The session that holds this lock can read the table, but not write it.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:45]:
```suggestion
- The session that holds this lock can read the table, but cannot write it.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+- Multiple sessions can acquire a `READ` lock from the same table at the same time.
+- Other sessions can read the table without explicitly acquiring a `READ` lock.
+
+The `READ LOCAL` lock is only for syntax compatibility with MySQL and is not supported.
+
+`WRITE` lock:
+
+- The session that holds this lock can read and write the table.
+- Only the session that holds this lock can access the table. No other session can access it until the lock is released.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:54]:
```suggestion
- Only the session that holds this lock can access the table. No other sessions can access it until the lock is released.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+
+`WRITE LOCAL` lock:
+
+- The session that holds this lock can read and write the table.
+- Only the session that holds this lock can access the table. Other sessions can read the table, but not write it.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:59]:
```suggestion
- Only the session that holds this lock can access the table. Other sessions can read the table, but cannot write it.
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+- `INFORMATION_SCHEMA`
+- `PERFORMANCE_SCHEMA`
+- `METRICS_SCHEMA`
+- `mysql`
+
+## MySQL compatibility
+
+### Table lock acquisition
+
+- In TiDB, if session A has already held a table lock, an error is returned if session B attempts to write to the table. In MySQL, the write request of session B is blocked until session A releases the table lock. Requests for locking the table from other sessions are blocked until the current session releases the `WRITE` lock.
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:101]:
```suggestion
- In TiDB, if session A has already held a table lock, an error is returned if session B attempts to write to the table. In MySQL, the write request of session B is blocked until session A releases the table lock, and requests for locking the table from other sessions are blocked until the current session releases the `WRITE` lock.
```

**Review by hfxsd — APPROVED** [review]
Rest LGTM

**Pull request merged**